### ISSUE TITLE
NotImplementedError with scipy.stats Levy_Stable distribution

### ISSUE THREAD

**Issue by valmunos** [scipy/scipy#8263]
<<Please describe the issue in detail here, and for bug reports fill in the fields below.>>

I have been fitting different distributions to my dataset, sales2014.  Calling the .fit method on levy_stable produced the following NotImplementedError.  All the other distributions seem to work just fine, so unless I misunderstood this one, I don't think it's due to a bug in my code.  My dataset is a collection of integers between 0 and 13000, inclusive, and is stored in a pandas Series.

### Reproducing code example:
```
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt

xmin = 0
xmax = 1
sales2014 = np.random.random(100)

lnspc = np.linspace(xmin, xmax,len(sales2014))
lsa,lsb = stats.levy_stable.fit(sales2014)  
pdf_levystable = stats.levy_stable.pdf(lnspc, lsa, lsb)  
plt.plot(lnspc, pdf_levystable, label="LevyStable")
```

### Error message:
```
NotImplementedError                       Traceback (most recent call last)
<ipython-input-95-a0d04fa4f4a3> in <module>()
     38 #plt.plot(lnspc, pdf_genextreme, label="Beta")
     39 
---> 40 lsa,lsb = stats.levy_stable.fit(sales2014)
     41 pdf_levystable = stats.levy_stable.pdf(lnspc, lsa, lsb)
     42 plt.plot(lnspc, pdf_levystable, label="LevyStable")

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in fit(self, data, *args, **kwds)
   2183                                          'scale' in kwds):
   2184             # get distribution specific starting locations
-> 2185             start = self._fitstart(data)
   2186             args += start[Narg:-2]
   2187         loc = kwds.pop('loc', start[-2])

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _fitstart(self, data, args)
   2021         if args is None:
   2022             args = (1.0,)*self.numargs
-> 2023         loc, scale = self._fit_loc_scale_support(data, *args)
   2024         return args + (loc, scale)
   2025 

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _fit_loc_scale_support(self, data, *args)
   2235 
   2236         # Estimate location and scale according to the method of moments.
-> 2237         loc_hat, scale_hat = self.fit_loc_scale(data, *args)
   2238 
   2239         # Compute the support according to the shape parameters.

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in fit_loc_scale(self, data, *args)
   2296 
   2297         """
-> 2298         mu, mu2 = self.stats(*args, **{'moments': 'mv'})
   2299         tmp = asarray(data)
   2300         muhat = tmp.mean()

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in stats(self, *args, **kwds)
   1025             if 'm' in moments:
   1026                 if mu is None:
-> 1027                     mu = self._munp(1, *goodargs)
   1028                 out0 = default.copy()
   1029                 place(out0, cond, mu * scale + loc)

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _munp(self, n, *args)
    783         # Silence floating point warnings from integration.
    784         olderr = np.seterr(all='ignore')
--> 785         vals = self.generic_moment(n, *args)
    786         np.seterr(**olderr)
    787         return vals

/opt/conda/lib/python3.5/site-packages/numpy/lib/function_base.py in __call__(self, *args, **kwargs)
   2216             vargs.extend([kwargs[_n] for _n in names])
   2217 
-> 2218         return self._vectorize_call(func=func, args=vargs)
   2219 
   2220     def _get_ufunc_and_otypes(self, func, args):

/opt/conda/lib/python3.5/site-packages/numpy/lib/function_base.py in _vectorize_call(self, func, args)
   2285                       for _a in args]
   2286 
-> 2287             outputs = ufunc(*inputs)
   2288 
   2289             if ufunc.nout == 1:

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _mom1_sc(self, m, *args)
   1603 
   1604     def _mom1_sc(self, m, *args):
-> 1605         return integrate.quad(self._mom_integ1, 0, 1, args=(m,)+args)[0]
   1606 
   1607     def _pdf(self, x, *args):

/opt/conda/lib/python3.5/site-packages/scipy/integrate/quadpack.py in quad(func, a, b, args, full_output, epsabs, epsrel, limit, points, weight, wvar, wopts, maxp1, limlst)
    321     if (weight is None):
    322         retval = _quad(func, a, b, args, full_output, epsabs, epsrel, limit,
--> 323                        points)
    324     else:
    325         retval = _quad_weight(func, a, b, args, full_output, epsabs, epsrel,

/opt/conda/lib/python3.5/site-packages/scipy/integrate/quadpack.py in _quad(func, a, b, args, full_output, epsabs, epsrel, limit, points)
    386     if points is None:
    387         if infbounds == 0:
--> 388             return _quadpack._qagse(func,a,b,args,full_output,epsabs,epsrel,limit)
    389         else:
    390             return _quadpack._qagie(func,bound,infbounds,args,full_output,epsabs,epsrel,limit)

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _mom_integ1(self, q, m, *args)
   1600     # moment calculated using ppf
   1601     def _mom_integ1(self, q, m, *args):
-> 1602         return (self.ppf(q, *args))**m
   1603 
   1604     def _mom1_sc(self, m, *args):

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in ppf(self, q, *args, **kwds)
   1913             goodargs = argsreduce(cond, *((q,)+args+(scale, loc)))
   1914             scale, loc, goodargs = goodargs[-2], goodargs[-1], goodargs[:-2]
-> 1915             place(output, cond, self._ppf(*goodargs) * scale + loc)
   1916         if output.ndim == 0:
   1917             return output[()]

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _ppf(self, q, *args)
    900 
    901     def _ppf(self, q, *args):
--> 902         return self._ppfvec(q, *args)
    903 
    904     def _isf(self, q, *args):

/opt/conda/lib/python3.5/site-packages/numpy/lib/function_base.py in __call__(self, *args, **kwargs)
   2216             vargs.extend([kwargs[_n] for _n in names])
   2217 
-> 2218         return self._vectorize_call(func=func, args=vargs)
   2219 
   2220     def _get_ufunc_and_otypes(self, func, args):

/opt/conda/lib/python3.5/site-packages/numpy/lib/function_base.py in _vectorize_call(self, func, args)
   2285                       for _a in args]
   2286 
-> 2287             outputs = ufunc(*inputs)
   2288 
   2289             if ufunc.nout == 1:

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _ppf_single(self, q, *args)
   1576         if not left:  # i.e. self.a = -inf
   1577             left = -1.*factor
-> 1578             while self._ppf_to_solve(left, q, *args) > 0.:
   1579                 right = left
   1580                 left *= factor

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _ppf_to_solve(self, x, q, *args)
   1564 
   1565     def _ppf_to_solve(self, x, q, *args):
-> 1566         return self.cdf(*(x, )+args)-q
   1567 
   1568     def _ppf_single(self, q, *args):

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in cdf(self, x, *args, **kwds)
   1740         if np.any(cond):  # call only if at least 1 entry
   1741             goodargs = argsreduce(cond, *((x,)+args))
-> 1742             place(output, cond, self._cdf(*goodargs))
   1743         if output.ndim == 0:
   1744             return output[()]

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _cdf(self, x, *args)
   1616 
   1617     def _cdf(self, x, *args):
-> 1618         return self._cdfvec(x, *args)
   1619 
   1620     ## generic _argcheck, _logcdf, _sf, _logsf, _ppf, _isf, _rvs are defined

/opt/conda/lib/python3.5/site-packages/numpy/lib/function_base.py in __call__(self, *args, **kwargs)
   2216             vargs.extend([kwargs[_n] for _n in names])
   2217 
-> 2218         return self._vectorize_call(func=func, args=vargs)
   2219 
   2220     def _get_ufunc_and_otypes(self, func, args):

/opt/conda/lib/python3.5/site-packages/numpy/lib/function_base.py in _vectorize_call(self, func, args)
   2285                       for _a in args]
   2286 
-> 2287             outputs = ufunc(*inputs)
   2288 
   2289             if ufunc.nout == 1:

/opt/conda/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py in _cdf_single(self, x, *args)
   1613 
   1614     def _cdf_single(self, x, *args):
-> 1615         return integrate.quad(self._pdf, self.a, x, args=args)[0]
   1616 
   1617     def _cdf(self, x, *args):

/opt/conda/lib/python3.5/site-packages/scipy/integrate/quadpack.py in quad(func, a, b, args, full_output, epsabs, epsrel, limit, points, weight, wvar, wopts, maxp1, limlst)
    321     if (weight is None):
    322         retval = _quad(func, a, b, args, full_output, epsabs, epsrel, limit,
--> 323                        points)
    324     else:
    325         retval = _quad_weight(func, a, b, args, full_output, epsabs, epsrel,

/opt/conda/lib/python3.5/site-packages/scipy/integrate/quadpack.py in _quad(func, a, b, args, full_output, epsabs, epsrel, limit, points)
    388             return _quadpack._qagse(func,a,b,args,full_output,epsabs,epsrel,limit)
    389         else:
--> 390             return _quadpack._qagie(func,bound,infbounds,args,full_output,epsabs,epsrel,limit)
    391     else:
    392         if infbounds != 0:

/opt/conda/lib/python3.5/site-packages/scipy/stats/_continuous_distns.py in _pdf(self, x, alpha, beta)
   3021 
   3022     def _pdf(self, x, alpha, beta):
-> 3023         raise NotImplementedError
   3024 levy_stable = levy_stable_gen(name='levy_stable')
   3025 

NotImplementedError: 
```

### Scipy/Numpy/Python version information:
```
0.19.0 1.11.3 sys.version_info(major=3, minor=5, micro=2, releaselevel='final', serial=0)
```

**Comment by rgommers** [comment]
Thanks @valmunos. I've edited your example to make it self-contained. I can reproduce the issue, is indeed a bug.

**Comment by valmunos** [comment]
Hey! Sorry to bug you ;) Is there any chance this fix will be implemented in the next five days?

**Comment by rgommers** [comment]
No problem to ask:) It's not very high on my list, so unless someone else has a go at fixing this, no.

**Comment by jcalbert** [comment]
@valmunos I think the problem is that the PDF does not exist except for special values of `alpha`.  Probably there should be a more descriptive error message / docstring, but it definitely won't get 'fixed' in the sense that you'll be able to use it soon.

**Comment by valmunos** [comment]
No worries, I have another package I can use in the meantime (PyLevy).
Thanks for the help though!

**Comment by jcalbert** [comment]
At some point I'll try to make a PR that gives a more descriptive error / docstring than the current

> Notes: Levy-stable distribution (only random variates available – ignore other docs)

**Comment by chrisb83** [comment]
I think this issue can be closed in view of #7374.
